rest-api/dao/participant_dao.py

```diff
@@ -201,6 +202,25 @@ def from_client_json(self, resource_json, id_=None, expected_version=None, clien
         withdrawalStatus=resource_json.get('withdrawalStatus'),
         suspensionStatus=resource_json.get('suspensionStatus'))
 
+  def add_missing_hpo_from_site(self, session, participant_id, site_id):
+    if site_id is None:
+      raise BadRequest('No site ID given for auto-pairing participant.')
+    site = SiteDao().get_with_session(session, site_id)
+    if site is None:
+      raise BadRequest('Invalid siteId reference %r.' % site_id)
+
+    participant = self.get_for_update(session, participant_id)
+    if participant is None:
+      raise BadRequest('No participant %r for HPO ID udpate.' % participant_id)
+    if participant.hpoId != UNSET_HPO_ID:
+      return
+
+    participant.hpoId = site.hpoId
+    participant.providerLink = make_primary_provider_link_for_id(site.hpoId)
+    if participant.participantSummary is None:
+      raise RuntimeError('No ParticipantSummary available for P%d.' % participant_id)
+    participant.participantSummary.hpoId = site.hpoId
```

Comment:
Do you need to do session.merge() somewhere? (I guess not?)

Reply:
I mostly didn't do it because it worked. (: But revisiting the docs, merging is only necessary when you create/modify something from outside a session. (Essentially an INSERT OR UPDATE I think.) http://docs.sqlalchemy.org/en/latest/orm/session_state_management.html#unitofwork-merging